base/file.jl

```diff
@@ -70,7 +70,16 @@ end
 
 # The following use Unix command line facilites
 
-cp(src::AbstractString, dst::AbstractString) = FS.sendfile(src, dst)
+function cp(src::AbstractString, dst::AbstractString; recursive::Bool=false)
+    if islink(src) || !isdir(src)
+        FS.sendfile(src, dst)
+    elseif recursive
+        mkdir(dst)
+        for p in readdir(src)
+            cp(joinpath(src, p), joinpath(dst, p), recursive=recursive)
+        end
+    end
```

Comment:
We should probably have an `else` here that throws an error if the user tries to copy a directory without `recursive`.